docs/DeveloperGuide.md

```diff
@@ -1,38 +1,254 @@
+application. After which, it will output a bye message and exits the application safely.
+
+### Command Component
+
+![Command.png](uml/Command.png)
+
+The command consists of three components: Enum CommandEnum, Abstract class Command and Class CommandParser. Each of them
+plays a role in retrieving the commands input by the user and redirecting to the correct components.
+
+#### CommandEnum
+
+The CommandEnum Enum contains all the commands that are available in the application.
+These are the only commands that can be input in the command line.
+
+#### Command
+
+The abstract class Command extends to several commands (eg `BudgetCommand`, `DepositCommand`, `ExpenseCommand`) and contains abstract methods execute() and isExit().
+The execute() method is executed from main, where it will go to the action class (eg `BudgetAction`) and run the requested action such as `add`.
+After execute() from main, isExit() controls the termination of BudgetBuddy by returning a boolean.
+
+#### CommandParser
+
+![CommandParserSequence.png](uml/CommandParserSequence.png)
+
+The `CommandParser` class takes in an input from the `Duke` class as seen in the sequence diagram above. It then parses
+the information
+to create a command of a specific type (eg `BudgetCommand`, `DepositCommand` , explained in the class diagram above)
+based
+the input.
 
 ## Product scope
-### Target user profile
 
-{Describe the target user profile}
+### Item Component
+
+The main 3 classes of Duke are the `budget` , `expense` and `deposit` class. Users are able to add, store and visualise
+the date relate to each
+of these classes. Each of these classes are modelled as an `Item`.
+
+![ItemClasses.png](uml/ItemClasses.png)
+
+### Budget Component
+
+#### BudgetCommand Class
+
+The `BudgetCommand` class contains methods that are related to the `Budget` function of Duke. Users are able to create
+new budgets, which are stored in a budget list. The category word from the user's input is taken from the first word of
+the users input,
+and the second word of the users input is the action word. The first word is processed through the `CommandParser`
+class. If the
+Command word is "budget", it will be processed into the `BudgetCommand` class. The action word is then processed into
+the `BudgetAction` class.
+The class diagram below shows how the `BudgetCommand` parent class is implemented, as well as its extended classes.
+
+![BudgetCommand.png](uml/BudgetCommand.png)
+
+#### BudgetCommand Sequence
+
+Once a string `input` from the user has been deemed as a `Budget` command as explained in the `CommandParser` sequence
+diagram above,
+the `input` will be passed into the `BudgetCommand` class. This class determines what budget commands to carry out to
+update the `data` stored in the `Budget` and `Expense` array lists, as shown in the sequence diagram below.
+
+![BudgetCommandSequence.png](uml/BudgetCommandSequence.png)
```

Comment:
It could be easier to read details if you could increase the definition of the picture.
![image](https://user-images.githubusercontent.com/122423020/228766291-a09a0a87-1a21-47c3-b91b-18777f38abe2.png)